whisperv6/whisper.go

```diff
@@ -1062,7 +1073,12 @@ func (whisper *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 
 				log.Info("received sync response", "count", len(resp.Envelopes), "final", resp.Final, "err", resp.Error, "cursor", resp.Cursor)
 
-				for _, envelope := range resp.Envelopes {
+				for _, rawEnvelope := range resp.Envelopes {
```

Comment:
it isn't `Envelope`s it receives now, right. also, won't it break compatibility? maybe we need a new opcode there?

Reply:
It would break compatibility, but it's only used on our fleet as I understand (so we can deploy them all at once and there should not be much problems as I understand), or is it used by others as well?

Reply:
but I thought this code is on a client, and there are some clients that won't be able to decode it...

Reply:
afaik this response is used only for synchronizing mail servers. pretty sure that nobody except us is using it.

i think if you will encode 
```
struct {
	Envelopes []rlp.RawValue
	Cursor    []byte
	Final     bool // if true it means all envelopes were processed
	Error     string
}
```

you will be able to decode into

```
type SyncResponse struct {
	Envelopes []*Envelope
	Cursor    []byte
	Final     bool // if true it means all envelopes were processed
	Error     string
}
```

Reply:
I believe this is only for Syncing between mailservers, syncing with clients should work as expected (I have deployed this code on the `test` fleet and I can pull from the mailserver), @adambabik do you know?

Reply:
@cammellos i think @adambabik is on vacation. i mentioned that this is a response for sync request, and sync request is not sent by a regular status client.

it is indeed was developed for syncing mail servers, and can be used only manually.

but also you shouldn't need this change if data will be encoded properly on a sender side

Reply:
@dshulyak sorry I missed your last message, I have updated the code, is that what you mean?

basically two I use 2 different structs, one for sending, which uses []rlp.RawValue which is then unmarshaled in []*Envelope

Reply:
yes, thank you :+1: